inference-engine/tests/unit/cpu/mkldnn_memory_desc_test.cpp

```diff
@@ -45,6 +45,384 @@ TEST(MemDescTest, Conversion) {
+                                                                             blk_desc_ptr->getBlockDims(), blk_desc_ptr->getOrder(),
+                                                                             blk_desc_ptr->getOffsetPadding(), blk_desc_ptr->getOffsetPaddingToData(),
+                                                                             blk_desc_ptr->getStrides());
+        return MemoryDescUtils::convertToDnnlMemoryDesc(cpu_blk_tdesc)->getDnnlDesc() ==
+               MemoryDescUtils::convertToDnnlMemoryDesc(blk_desc_ptr)->getDnnlDesc();
+    };
+
+    std::pair<dnnl::memory::format_tag, dnnl::memory::dims> payload[] {
+            { dnnl::memory::format_tag::nChw16c,     {1, 1, 10, 10} },  // auto blocked
+            { dnnl::memory::format_tag::nhwc,        {4, 2, 10, 7 } },  // permuted
+            { dnnl::memory::format_tag::nchw,        {4, 2, 10, 7 } },  // plain
+            { dnnl::memory::format_tag::NChw16n16c,  {4, 2, 10, 7 } },  // blocked for 2 dims
+            { dnnl::memory::format_tag::BAcd16a16b,  {4, 2, 10, 7 } },  // blocked and permuted outer dims
+            { dnnl::memory::format_tag::Acdb16a,     {96, 1, 7, 7 } },  // same strides but not default order
+    };
+
+    for (const auto &p : payload) {
+        ASSERT_TRUE(converted_correctly_BlockedMemoryDescPtr(p.first, p.second));
+        ASSERT_TRUE(converted_correctly_DnnlMemoryDescPtr(p.first, p.second));
+    }
+}
+
+TEST(MemDescTest, ConversionFormatKindUndef) {
+    auto isCompatibleDnnlBlockedMemoryDesc = [] (dnnl::memory::format_tag fmt, const dnnl::memory::dims& dims) {
+        dnnl::memory::desc orig_desc {dims, dnnl::memory::data_type::u8, fmt};
+        orig_desc.data.format_kind = dnnl_format_kind_undef;
+        DnnlMemoryDescPtr dnnl_desc_ptr = MKLDNNExtensionUtils::makeDescriptor(orig_desc);
+        DnnlBlockedMemoryDesc dnnl_blk_desc = MemoryDescUtils::convertToDnnlBlockedMemoryDesc(*dnnl_desc_ptr);
+    };
+
+    std::pair<dnnl::memory::format_tag, dnnl::memory::dims> payload[] {
+            { dnnl::memory::format_tag::nChw16c,     {1, 1, 10, 10} },  // auto blocked
+            { dnnl::memory::format_tag::nhwc,        {4, 2, 10, 7 } },  // permuted
+            { dnnl::memory::format_tag::nchw,        {4, 2, 10, 7 } },  // plain
+            { dnnl::memory::format_tag::NChw16n16c,  {4, 2, 10, 7 } },  // blocked for 2 dims
+            { dnnl::memory::format_tag::BAcd16a16b,  {4, 2, 10, 7 } },  // blocked and permuted outer dims
+            { dnnl::memory::format_tag::Acdb16a,     {96, 1, 7, 7 } },  // same strides but not default order
+    };
+
+    for (const auto &p : payload) {
+        ASSERT_ANY_THROW(isCompatibleDnnlBlockedMemoryDesc(p.first, p.second));
+    }
+}
+
+TEST(MemDescTest, isCompatible) {
+    auto isCompatibleDnnlMemoryDesc = [] (dnnl::memory::format_tag fmt, const dnnl::memory::dims& dims) {
```

Comment:
i guess we also need to add some test cases for dynamic shapes when after conversion we create new desc with difference but compatible shape

let's discuss it later